### DISCUSSION TITLE
The UI Schema for myschema does not exist.

### DISCUSSION THREAD

**Discussion by stefan-korn** [GetDKAN/dkan#4030] [Q&A]
The Drupal metastore Schema Retriever has valid schema ids hardcoded in its "getAllIds()" method.

Is there any supported way to have your own schema ids added there? So you can put schema definitions of your own in the schema/collections folder and use them? Wouldn't it be something that users of DKAN2 might want to do, Maybe via configuration?

One could possibly swap or decorate the metastore retriever service, but was wondering whether there is something easier?

Thinking aloud: Why not skipping this getAllIds() method at all and leave it up to the user which schema definitions to use?

**Comment by dafeder** [comment]
@stefan-korn yes, agreed that this is a little weird. Have been considering defining all schemas via a YML file similar to services, that could be overridden in a custom module. But yes an easy fix would probably be to scan the folder and take the names from that. I think the hardcoded list was supposed to be temporary and was just never revisited.

I think you do need a getAllIds() method for places in the codebase where it needs a full list of available schemas. It should just be dynamic in some way.
